pkg/providers/vsphere/virtualmachine/configspec.go

```diff
@@ -115,9 +115,11 @@ func CreateConfigSpec(
 		}
 	}
 
-	// Initially set the guest ID in ConfigSpec to ensure VM is created with the expected guest ID.
+	// VM Class specified guestID is not supported. Ignore if set.
+	configSpec.GuestId = ""
```

Comment:
Does it make more sense to do this in `SanitizeVMClassConfigSpec()`?

Reply:
you're right. It does get called before the CreateConfigSpec(). Will move it there. Thanks BV.